src/azure-cli/azure/cli/command_modules/resource/custom.py

```diff
@@ -2120,6 +2282,377 @@ def list_template_specs(cmd, resource_group_name=None, name=None):
+    rcf = _resource_deploymentstacks_client_factory(cmd.cli_ctx)
+
+    delete_resources_enum, delete_resource_groups_enum = _prepare_stacks_delete_detach_models(
+        rcf, delete_all, delete_resource_groups, delete_resources)
+    deny_settings_enum = _prepare_stacks_deny_settings(rcf, deny_settings_mode)
+
+    excluded_principals_array = _prepare_stacks_excluded_principals(deny_settings_excluded_principals)
+    excluded_actions_array = _prepare_stacks_excluded_actions(deny_settings_excluded_actions)
+
+    tags = tags or {}
+
+    if query_string and not template_uri:
+        raise IncorrectUsageError('please provide --template-uri if --query-string is specified')
+
+    if [template_file, template_spec, template_uri].count(None) != 2:
+        raise InvalidArgumentValueError(
+            "Please enter only one of the following: template file, template spec, or template url")
+    try:
+        get_subscription_response = rcf.deployment_stacks.get_at_subscription(name)
+        if get_subscription_response:
+            if get_subscription_response.location != location:
+                raise CLIError("Cannot change location of an already existing stack at subscription scope.")
+            # bypass if yes flag is true
+            _build_stacks_confirmation_string(rcf, yes, name, delete_resources_enum, delete_resource_groups_enum)
+    except:  # pylint: disable=bare-except
+        pass
+
+    if not deployment_resource_group:
+        deployment_scope = "/subscriptions/" + get_subscription_id(cmd.cli_ctx)
+    else:
+        deployment_scope = "/subscriptions/" + \
+            get_subscription_id(cmd.cli_ctx) + "/resourceGroups/" + deployment_resource_group
+
+    action_on_unmanage_model = rcf.deployment_stacks.models.DeploymentStackPropertiesActionOnUnmanage(
+        resources=delete_resources_enum, resource_groups=delete_resource_groups_enum)
+    apply_to_child_scopes = deny_settings_apply_to_child_scopes
+    deny_settings_model = rcf.deployment_stacks.models.DenySettings(
+        mode=deny_settings_enum, excluded_principals=excluded_principals_array, excluded_actions=excluded_actions_array, apply_to_child_scopes=apply_to_child_scopes)
+    deployment_stack_model = rcf.deployment_stacks.models.DeploymentStack(
+        description=description, location=location, action_on_unmanage=action_on_unmanage_model, deployment_scope=deployment_scope, deny_settings=deny_settings_model, tags=tags)
+
+    deployment_stack_model = _prepare_stacks_templates_and_parameters(
+        cmd, rcf, deployment_stack_model, template_file, template_spec, template_uri, parameters, query_string)
+
+    return sdk_no_wait(False, rcf.deployment_stacks.begin_create_or_update_at_subscription, name, deployment_stack_model)
```

Comment:
Currently, these methods do not need to support for `--no-wait`, right?

Reply:
It's been a while since these were added but I believe these are required

Reply:
Yes, we can support it in a future PR